controllers/callhomeconfig_controller.go

```diff
@@ -139,41 +143,40 @@ func (r *CallHomeConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			fmt.Sprintf("unexpected error getting service account: %s", err),
 			ctrl.Result{})
 	}
-	if len(sa.Secrets) < 1 {
-		return retryLater(r, log, &chc, "serviceaccount",
-			namespacedName.Name, "does not have a token")
-	}
-	tokenSecretName := sa.Secrets[0]
-	var token corev1.Secret
+	var kubeconfigSecret corev1.Secret
 	if err := r.Get(ctx, types.NamespacedName{
 		Namespace: req.Namespace,
-		Name:      tokenSecretName.Name,
-	}, &token); err != nil {
+		Name:      "cluster-autoscaler-mgmt-kubeconfig",
```

Comment:
It is no longer necessary to read the service account, so you can:
- Delete lines 132-145
- Delete the corresponding RBAC rules for reading service accounts

Furthermore, you should not read a hard-coded token name like `cluster-autoscaler-mgmt-kubeconfig`. The name should come from the CHC resource. So there are two choices:
1. Preserve existing API compatiblity: List all secrets in the namespace, and find the one that has an annotation that matches `chc.Spec.ServiceAccountName` (my preferred way)
2. Break existing API compatibility, and rename `chc.Spec.ServiceAccountName` to `chc.Spec.ServiceAccountTokenName`, and use that to read the right secret. The manifest for the CHC should set that field to `cluster-autoscaler-mgmt-kubeconfig`. This is probably ok because I don't think any users/customers are using CHC and CAS yet.
3. Introduce a new API version of the CHC.

Reply:
The service account is not read. However, it's presence is necessary as k8s control plane will not create the secret unless the service account is present. Also, if the service account is deleted, the control plane cleans up the associated secret as well. This behaviour was tested by us on a dummy cluster with k8s version v1.21.14 where we create a secret with the annotation(`kubernetes.io/service-account.name: xyz`) but don't create the service account, the secret create request itself succeeds, but the secret is instantly deleted by the control plane. The same behaviour is observed in k8s version 1.24 and 1.25 as well for the management cluster.

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-an-api-token-for-a-serviceaccount

Reply:
Sure, we'll opt for the path of least resistance and go with
> Preserve existing API compatiblity: List all secrets in the namespace, and find the one that has an annotation that matches chc.Spec.ServiceAccountName (my preferred way)